docs/content/latest/explore/ysql-language-features/gin.md

```diff
@@ -0,0 +1,501 @@
+
+The assumption in the following examples is that the user is using the GIN index in ways that take advantage of it.
+
+First, enable timing for future queries.
+
+```sql
+\timing on
+```
+
+First, test GIN index on tsvector:
+
+```
+SET enable_indexscan = off;
+SELECT * FROM vectors WHERE v @@ to_tsquery('simple', 'the');
+-- Run it several times to reduce cache bias.
+SELECT * FROM vectors WHERE v @@ to_tsquery('simple', 'the');
+SELECT * FROM vectors WHERE v @@ to_tsquery('simple', 'the');
+```
+
+```output
+                  v                  | k
+-------------------------------------+---
+ 'brown':3 'fox':4 'quick':2 'the':1 | 1
+ 'jumps':1 'over':2 'the':3          | 2
+(2 rows)
+
+Time: 11.141 ms
+```
+
+```sql
+SET enable_indexscan = on;
+SELECT * FROM vectors WHERE v @@ to_tsquery('simple', 'the');
+-- Run it several times to reduce cache bias.
+SELECT * FROM vectors WHERE v @@ to_tsquery('simple', 'the');
+SELECT * FROM vectors WHERE v @@ to_tsquery('simple', 'the');
+```
+
+```output
+...
+Time: 2.838 ms
+```
+
+Notice the over 3x timing improvement when using GIN index.
+This is on a relatively small table: a little over 1000 rows.
+With more and/or bigger rows, the timing improvement should get better.
+
+Next, int array:
+
+```sql
+SET enable_indexscan = off;
+SELECT * FROM arrays WHERE a @> '{6}';
+SELECT * FROM arrays WHERE a @> '{6}';
+SELECT * FROM arrays WHERE a @> '{6}';
+```
+
+```output
+    a    | k
+---------+---
+ {1,1,6} | 1
+ {1,6,1} | 2
+ {2,3,6} | 3
+(3 rows)
+
+Time: 9.501 ms
+```
+
+```sql
+SET enable_indexscan = on;
+SELECT * FROM arrays WHERE a @> '{6}';
+SELECT * FROM arrays WHERE a @> '{6}';
+SELECT * FROM arrays WHERE a @> '{6}';
+```
+
+```output
+...
+Time: 2.989 ms
+```
+
+Next, jsonb:
+
+```sql
+SET enable_indexscan = off;
+SELECT * FROM jsonbs WHERE j ? 'some';
+SELECT * FROM jsonbs WHERE j ? 'some';
+SELECT * FROM jsonbs WHERE j ? 'some';
+```
+
+```output
+                                         j                                          | k
+------------------------------------------------------------------------------------+---
+ {"some": ["where", "how"]}                                                         | 5
+ {"and": ["another", "element", "not", "a", "number"], "some": {"nested": "jsonb"}} | 6
+ {"some": "thing"}                                                                  | 4
+ {"some": "body"}                                                                   | 2
+ {"some": "one"}                                                                    | 3
+(5 rows)
+
+Time: 13.451 ms
+```
+
+```sql
+SET enable_indexscan = on;
+SELECT * FROM jsonbs WHERE j ? 'some';
+SELECT * FROM jsonbs WHERE j ? 'some';
+SELECT * FROM jsonbs WHERE j ? 'some';
+```
+
+```output
+...
+Time: 2.115 ms
+```
+
+### Unsupported queries
+
+Sometimes, an unsupported query may be encountered by getting an ERROR.
+Here, I show workarounds to some of these cases.
```

Comment:
nit: 
```
Below are some workarounds for some of these cases.
```
Also, up to you, but if it makes sense and is easy, we can maybe link to the relevant issues for each of the cases below. This way folks that hit/want this feature can post or upvote on the GH issue.